src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/Microsoft/Build/Tasks/Windows/CreateTemporaryTargetAssemblyProject.cs

```diff
@@ -0,0 +1,333 @@
+        { get; set; }
+
+        /// <summary>
+        /// ReferencePath
+        ///    A list of resolved reference assemblies.
+        ///    The list will replace the original Reference item list in generated project file.
+        /// </summary>
+        public ITaskItem[] ReferencePath
+        { get; set; }
+
+        /// <summary>
+        /// ReferencePathTypeName
+        ///   The appropriate item name which is used to keep the Reference list in managed compiler task.
+        ///   It is "ReferencePath" for now.
+        ///   
+        ///   Adding this property is to make the type name configurable, if it is changed, 
+        ///   No code is required to change in this task, but set a new type name in project file.
+        /// </summary>
+        [Required]
+        public string ReferencePathTypeName
+        { get; set; }
+
+        [Output]
+        public string TemporaryTargetAssemblyProjectName 
+        { get; set; }
+
+        //
+        // Remove specific items from project file. Every item should be under an ItemGroup.
+        //
+        private void RemoveItemsByName(XmlDocument xmlProjectDoc, string sItemName)
+        {
+            if (xmlProjectDoc == null || String.IsNullOrEmpty(sItemName))
+            {
+                // When the parameters are not valid, simply return it, instead of throwing exceptions.
+                return;
+            }
+
+            //
+            // The project file format is always like below:
+            //
+            //  <Project  xmlns="...">
+            //     <ProjectGroup>
+            //         ......
+            //     </ProjectGroup>
+            //
+            //     ...
+            //     <ItemGroup>
+            //         <ItemNameHere ..../>
+            //         ....
+            //     </ItemGroup>
+            //
+            //     ....
+            //     <Import ... />
+            //     ...
+            //     <Target Name="xxx" ..../>
+            //     
+            //      ...
+            //
+            //  </Project>
+            //
+            //
+            // The order of children nodes under Project Root element is random
+            //
+
+            XmlNode root = xmlProjectDoc.DocumentElement;
+
+            if (root.HasChildNodes == false)
+            {
+                // If there is no child element in this project file, just return immediatelly.
+                return;
+            }
+
+            for (int i = 0; i < root.ChildNodes.Count; i++)
+            {
+                XmlElement nodeGroup = root.ChildNodes[i] as XmlElement;
+
+                if (nodeGroup != null && String.Compare(nodeGroup.Name, ITEMGROUP_NAME, StringComparison.OrdinalIgnoreCase) == 0)
+                {
+                    //
+                    // This is ItemGroup element.
+                    //
+                    if (nodeGroup.HasChildNodes)
+                    {
+                        ArrayList itemToRemove = new ArrayList();
```

Comment:
It seems that we can use `List<XmlElement>, which can avoid the judgment on the [244](https://github.com/dotnet/wpf/pull/3585/files#diff-18041220a9709cd678f330094a6fbbe9R244) line